File: BankImport/BankImport/Slf.Dms.Service.BankImport.Drg.Util/TextBoxTraceListener.vb
Imports System
Imports System.Diagnostics
Imports System.Windows.Forms

Namespace Slf.Dms.Service.BankImport.Drg.Util
    Public Class TextBoxTraceListener
        Inherits TraceListener
        ' Methods
        Public Sub New(ByVal output As TextBox)
            Me._output = output
        End Sub

        Public Overrides Sub Write(ByVal message As String)
            If Not String.IsNullOrEmpty(message) Then
                Me._output.AppendText(message)
                Me._output.SelectionStart = Me._output.Text.Length
                Me._output.ScrollToCaret
            End If
        End Sub

        Public Overrides Sub WriteLine(ByVal message As String)
            Me.WriteIndent
            Me.Write((message & Environment.NewLine))
        End Sub


        ' Fields
        Private _output As TextBox
    End Class
End Namespace

File: Slf.Dms.Service.ICBImport/TextBoxTraceListener.cs
using System;
using System.Diagnostics;
using System.Text;
using System.Windows.Forms;

namespace Drg.Util
{
	/// <summary>
	/// Summary description for TextBoxTraceListener.
	/// </summary>
	public class TextBoxTraceListener : TraceListener
	{
		TextBox _output;

		public TextBoxTraceListener(TextBox output)
		{
			_output = output;
		}

		//StringBuilder sb = new StringBuilder();

		public override void Write(string message)
		{
            if (message != null)
            {
                _output.AppendText(message);
                _output.SelectionStart = _output.Text.Length;
                _output.ScrollToCaret();
            }
			//sb.Append(message);
		}

		public override void WriteLine(string message)
		{
			WriteIndent();
			Write(message);
            Write(Environment.NewLine);
		}

	}
}
